src/spider/storage/DataStorage.hpp

```diff
@@ -32,6 +32,36 @@ class DataStorage {
             = 0;
     virtual auto get_data(StorageConnection& conn, boost::uuids::uuid id, Data* data) -> StorageErr
             = 0;
+    /**
+     * Get data and register data reference from a driver in a single transaction.
+     * @param conn
+     * @param driver_id
+     * @param data_id
+     * @param data output data
+     * @return StorageErr::Success if the transaction succeed. Error types otherwise.
+     */
+    virtual auto get_data_driver(
```

Comment:
Double check I didn't miss this change anywhere.
```suggestion
    virtual auto get_driver_data(
```